File: src/code/java/Q461.java
package code.java;

public class Q461 {
    // 0 <= x, y < 2^31
    public int hammingDistance(int x, int y) {
        int count = 0;
        int n = x ^ y;
        while (n > 0) {
            count += (n & 1);
            n = n >>> 1;
        }
        return count;
    }
}

File: src/code/js/Q461.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function (x, y) {
    var count = 0, n = x ^ y;
    while (n > 0) {
        count += (n & 1);
        n = n >> 1;
    }
    return count;
};